main.go

```diff
@@ -71,17 +118,30 @@ func main() {
 		log.Infof("Uploading file %s", cpuFile)
 		cpuResponse, err := uploadFile(envVars.UploadAPIURL, envVars.Token, cpuValues)
 		if err != nil {
-			log.WithError(err).Errorf("Failed to upload file %s", cpuFile)
+			return errors.Errorf("Failed to upload file %s", cpuFile)
 		}
 		uploads = append(uploads, cpuResponse.FileInfos[0].ID)
 	}
-
 	log.Info("Posting files")
-	err = postFile(envVars.PostAPIURL, envVars.ChannelID, envVars.Token, uploads, envVars.MattermostProfileTargets)
+	err := postFile(envVars.PostAPIURL, envVars.ChannelID, envVars.Token, deployment, uploads)
+	if err != nil {
+		return errors.Errorf("Failed to post files")
```

Comment:
better to wrap the error `errors.Wrap(err, "failed to post files")`

Reply:
Thanks fixed